packages/mobile/src/firebase/firebase.ts

```diff
@@ -21,6 +21,10 @@ export const initializeAuth = async (app: Firebase, address: string) => {
   await userRef.child(user.user.uid).transaction((userData) => {
     if (userData == null) {
       return { address }
+    } else if (userData.address !== undefined && userData.address !== address) {
+      // This shouldn't happen! If this is thrown it means the firebase user is reused
+      // with different addresses (which we don't want) or the db was incorrectly changed remotely!
+      throw new Error("User address in the db doesn't match persisted address")
```

Comment:
Just want to confirm that it was intentional to throw here instead of returning undefined? This callback is what will run on the db to modify the data. I don't think the error will propagate to the app, but I could be wrong. I'm uncertain what happens if we throw here.

Reply:
Good question! 
It triggers a red screen in dev, and it's not captured by the saga promise flow. 
It's logged by Sentry and crashes the app. Which is what I intended as is I see it like a logic error which shouldn't happen.